python/cudf_polars/cudf_polars/dsl/expr.py

```diff
@@ -1188,7 +1188,11 @@ class Cast(Expr):
     def __init__(self, dtype: plc.DataType, value: Expr) -> None:
         super().__init__(dtype)
         self.children = (value,)
-        if not plc.unary.is_supported_cast(self.dtype, value.dtype):
+        if (
+            not plc.traits.is_fixed_width(self.dtype)
+            and plc.traits.is_fixed_width(value.dtype)
```

Comment:
This logic looks wrong to me. 

The supported casts are: `is_fixed_width(from) and is_fixed_width(to) and is_supported_cast(from, to)`, so you want to raise for the converse

```
if not (plc.traits.is_fixed_width(self.dtype) and plc.traits.is_fixed_width(value.dtype) and plc.unary.is_supported_cast(value.dtype, self.dtype):
    raise ...
```

Note also the change of order in `is_supported_cast`. `value` provides the source dtype and `self` provides the value dtype.

Reply:
Hmm perhaps I misunderstood, I thought this was allowing a cast from a non-fixed to fixed. Do we need both types to be fixed width?

Reply:
`cudf::cast` docstring says ` * Supports only fixed-width types.` But `is_supported_cast` only answers correctly if the inputs are fixed-width.

Reply:
I didn't realise this when I introduced the `cudf::is_supported_cast(from_dtype, to_dtype)` public function.